tests/test_multiaddr.py

```diff
@@ -218,3 +223,47 @@ def test_get_value():
     assert_value_for_proto(a, P_IP4, "0.0.0.0")
     assert_value_for_proto(a, P_UDP, "12345")
     assert_value_for_proto(a, P_UTP, "")
+
+
+def test_bad_initialization_no_params():
+    with pytest.raises(ValueError):
+        Multiaddr()
+
+
+def test_bad_initialization_too_many_params():
+    with pytest.raises(ValueError):
+        Multiaddr("/ip4/0.0.0.0", string_to_bytes("/ip4/0.0.0.0"))
+
+
+def test_get_value_too_many_fields_protocol(monkeypatch):
+    monkeypatch.setattr("multiaddr.multiaddr.Multiaddr.__str__",
+                        lambda ignore: '/udp/1234/5678')
+    a = Multiaddr("/ip4/127.0.0.1/udp/1234")
+    with pytest.raises(ValueError):
+        a.value_for_protocol(P_UDP)
+
+
+def test_multi_addr_str_corruption(monkeypatch):
+    a = Multiaddr("/ip4/127.0.0.1/udp/1234")
+    a._bytes = b"047047047"
+
+    with pytest.raises(ValueError):
+        str(a)
+
+
+def test_decapsulate_incorrect_byte_boundaries(monkeypatch):
```

Comment:
> I'm now wondering if this error message is even useful -- are there any valid code paths that can trigger it? Maybe it should just get deleted. What do you think?

The only path I can think of is passing in a non-Multiaddr object for the other param that actually contains part of the string representation of self.  This seems like user error to the extreme and a strange thing to check for.  Yes I agree it could be deleted.